### ISSUE TITLE
fetching a document which 401s

### ISSUE THREAD

**Issue by bblfish** [github/fetch#229]
I have a document served from localhost that calls a document `https://joe.example:8443/2013/card` on a different server that returns a 401 .

```http
$ curl -k -i https://joe.example:8443/2013/card
HTTP/1.1 401 Unauthorized
Access-Control-Allow-Origin: *
Content-Type: text/html; charset=utf-8
WWW-Authenticate: Signature realm="/"
Content-Length: 2374
...
```

Currently in ` Chrome  48.0.2554.0 canary (64-bit)` this shows up as `The FetchEvent for "https://joe.example:8443/2013/card" resulted in a network error response: the promise was rejected.`
If the same file were served by localhost on a different port it would be accessible.

So this seems to be a CORS problem. I am not sure if the spec allows me to capture 401s or not. I have I think set `{ mode: "cors", credentials: "omit" } ` ( I say I think because I am working from scala-js and am not yet that familiar with it. The console shows the following:

![screen shot 2015-11-05 at 00 10 24](https://cloud.githubusercontent.com/assets/124506/10956212/b9f15c64-8351-11e5-8409-d610432f73da.png)

It is not so clear from the fetch documentation what the different modes actually mean. Should I be able to find the 401? I'd like to be able to work out what authentication I can use on the future requests.

It could also be that I my RequestInit datastructure is incorrect. But as I don't yet know what to expect it is difficult to work out what may be wrong.

**Comment by dgraham** [comment]
- https://github.com/github/fetch/issues/181 CORS debugging tips
- https://github.com/github/fetch/issues/199 CORS authentication cookie settings
- https://github.com/github/fetch/issues/201 CORS response headers and 401 handling

**Comment by bblfish** [comment]
thanks @dgraham . I made two additional mistakes:
- I have a server running locally under `https://joe.example:8443` which I can access after editing `/etc/hosts` , but the certificate my server was running had a `CN=localhost`. I created a new certificate with  `CN=joe.example` and that helped
- I also had the `OPTIONS` request return a 401. It works when I return a `200 Ok` instead.

It may be worth closing such issues but tagging them "documentation" so that people writing more general public documentation can look through these questions and write HOWTOs with the answers given.
